app/presenters/stats_presenter.rb

```diff
@@ -0,0 +1,192 @@
+# frozen_string_literal: true
+
+class StatsPresenter # rubocop:disable Metrics/ClassLength
+  def initialize(user)
+    @user = user
+  end
+
+  def platform_stats
+    {
+      registered_users: User.count,
+      confirmed_users: User.confirmed.count,
+      participating_users: User.distinct(:id).joins(:completions).count,
+      users_with_snippets: User.distinct(:id).joins(:snippets).count,
+      total_snippets: Snippet.count,
+      gold_stars:,
+      silver_stars:,
+      daily_completers:,
+      users_per_star:,
+      user_solved_today: @user&.completions&.where(day: Aoc.latest_day)&.count
+    }
+  end
+
+  def achievements
+    {
+      fan: set_fan_achievement,
+      the_answer: set_the_answer_achievement,
+      doomed_sundays: set_doomed_sundays_achievement,
+      influencer: set_influencer_achievement,
+      the_godfather: set_the_godfather_achievement,
+      belonging: set_belonging_achievement,
+      mobster: set_mobster_achievement,
+      jedi_master: set_jedi_master_achievement,
+      madness: set_madness_achievement,
+      jeweler: set_jeweler_achievement,
+      snake_charmer: set_snake_charmer_achievement,
+      picasso: set_picasso_achievement
+    }
+  end
```

Comment:
I think this should be done dynamically with I18n for the translation and some logic on the model or smth. Let's go with this for now, I'll think of something later.

Reply:
Yeah when I see this I'm not _super_ happy, but each achievement has peculiarities that makes a general approach not straightforward to me.

What do you mean by translation?